File: src/main/java/com/radixpro/enigma/xchg/api/XchgApiInjector.java
/*
 * Jan Kampherbeek, (c) 2020.
 * Enigma is open source.
 * Please check the file copyright.txt in the root of the source for further details.
 *
 */

package com.radixpro.enigma.xchg.api;

import com.radixpro.enigma.be.calc.SeFrontend;
import com.radixpro.enigma.be.handlers.BeHandlersInjector;
import com.radixpro.enigma.be.persistency.BePersistencyInjector;

public class XchgApiInjector {

   public static AspectsApi injectAspectsApi() {
      return new AspectsApi(BeHandlersInjector.injectAspectsHandler());
   }

   public static CalculatedChartApi injectCalculatedChartApi() {
      return new CalculatedChartApi(BeHandlersInjector.injectCalculatedChartHandler());
   }

   public static DateTimeApi injectDateTimeApi() {
      return new DateTimeApi(SeFrontend.INSTANCE);
   }     // TODO move SeFrontend to scope

   public static MidpointsApi injectMidpointsApi() {
      return new MidpointsApi(BeHandlersInjector.injectMidpointsHandler());
   }

   public static InputDataFileApi injectInputDataFileApi() {
      return new InputDataFileApi(BeHandlersInjector.injectInputDataFileHandler());
   }

   public static PersistedChartDataApi injectPersistedChartDataApi() {
      return new PersistedChartDataApi(BePersistencyInjector.injectChartDataDao());
   }

   public static PersistedConfigurationApi injectPersistedConfigurationApi() {
      return new PersistedConfigurationApi(BePersistencyInjector.injectConfigurationDao());
   }

   public static PersistedDataFileApi injectPersistedDataFileApi() {
      return new PersistedDataFileApi(BeHandlersInjector.injectDataFileHandler());
   }

   public static PersistedPropertyApi injectPersistedPropertyApi() {
      return new PersistedPropertyApi(BePersistencyInjector.injectPropertyDao());
   }

   public static PrimaryApi injectPrimaryApi() {
      return new PrimaryApi(BeHandlersInjector.injectPrimaryHandler());
   }

   public static SecundaryApi injectSecundaryApi() {
      return new SecundaryApi(BeHandlersInjector.injectEphProgCalcHandler(), BeHandlersInjector.injectSecundaryDateHandler(),
            BeHandlersInjector.injectProgAspectHandler());
   }

   public static SolarReturnApi injectSolarReturnApi() {
      return new SolarReturnApi(BeHandlersInjector.injectSolarReturnHandler());
   }

   public static TetenburgApi injectTetenburgApi() {
      return new TetenburgApi(BeHandlersInjector.injectTetenburgHandler());
   }

   public static TransitsApi injectTransitsApi() {
      return new TransitsApi(BeHandlersInjector.injectEphProgCalcHandler(), BeHandlersInjector.injectProgAspectHandler());
   }

   public static VersionApi injectVersionApi() {
      return new VersionApi(BePersistencyInjector.injectVersionDao());
   }

}

File: src/main/java/com/radixpro/enigma/xchg/api/XchgApiInjector.kt
/*
 * Jan Kampherbeek, (c) 2020.
 * Enigma is open source.
 * Please check the file copyright.txt in the root of the source for further details.
 *
 */
package com.radixpro.enigma.xchg.api

import com.radixpro.enigma.be.handlers.BeHandlersInjector
import com.radixpro.enigma.be.handlers.BeHandlersInjector.injectAspectsHandler
import com.radixpro.enigma.be.handlers.BeHandlersInjector.injectCalculatedChartHandler
import com.radixpro.enigma.be.handlers.BeHandlersInjector.injectDataFileHandler
import com.radixpro.enigma.be.handlers.BeHandlersInjector.injectEphProgCalcHandler
import com.radixpro.enigma.be.handlers.BeHandlersInjector.injectInputDataFileHandler
import com.radixpro.enigma.be.handlers.BeHandlersInjector.injectMidpointsHandler
import com.radixpro.enigma.be.handlers.BeHandlersInjector.injectPrimaryHandler
import com.radixpro.enigma.be.handlers.BeHandlersInjector.injectProgAspectHandler
import com.radixpro.enigma.be.handlers.BeHandlersInjector.injectSecundaryDateHandler
import com.radixpro.enigma.be.handlers.BeHandlersInjector.injectSolarReturnHandler
import com.radixpro.enigma.be.handlers.BeHandlersInjector.injectTetenburgHandler
import com.radixpro.enigma.be.persistency.BePersistencyInjector

object XchgApiInjector {
    @JvmStatic
    fun injectAspectsApi(): AspectsApi {
        return AspectsApi(injectAspectsHandler())
    }

    @JvmStatic
    fun injectCalculatedChartApi(): CalculatedChartApi {
        return CalculatedChartApi(injectCalculatedChartHandler())
    }

    @JvmStatic
    fun injectDateTimeApi(): DateTimeApi {
        return DateTimeApi()
    }

    @JvmStatic
    fun injectMidpointsApi(): MidpointsApi {
        return MidpointsApi(injectMidpointsHandler())
    }

    @JvmStatic
    fun injectInputDataFileApi(): InputDataFileApi {
        return InputDataFileApi(injectInputDataFileHandler())
    }

    @JvmStatic
    fun injectPersistedChartDataApi(): PersistedChartDataApi {
        return PersistedChartDataApi(BePersistencyInjector.injectChartDataDao())
    }

    @JvmStatic
    fun injectPersistedConfigurationApi(): PersistedConfigurationApi {
        return PersistedConfigurationApi(BePersistencyInjector.injectConfigurationDao())
    }

    @JvmStatic
    fun injectPersistedDataFileApi(): PersistedDataFileApi {
        return PersistedDataFileApi(injectDataFileHandler())
    }

    @JvmStatic
    fun injectPersistedPropertyApi(): PersistedPropertyApi {
        return PersistedPropertyApi(BePersistencyInjector.injectPropertyDao())
    }

    @JvmStatic
    fun injectPrimaryApi(): PrimaryApi {
        return PrimaryApi(injectPrimaryHandler())
    }

    @JvmStatic
    fun injectSecundaryApi(): SecundaryApi {
        return SecundaryApi(injectEphProgCalcHandler(), injectSecundaryDateHandler(),
                injectProgAspectHandler())
    }

    fun injectSolarReturnApi(): SolarReturnApi {
        return SolarReturnApi(injectSolarReturnHandler())
    }

    @JvmStatic
    fun injectStatsProjApi(): StatsProjApi {
        return StatsProjApi(BeHandlersInjector.injectStatsProjHandler())
    }

    @JvmStatic
    fun injectTetenburgApi(): TetenburgApi {
        return TetenburgApi(injectTetenburgHandler())
    }

    @JvmStatic
    fun injectTransitsApi(): TransitsApi {
        return TransitsApi(injectEphProgCalcHandler(), injectProgAspectHandler())
    }

    fun injectVersionApi(): VersionApi {
        return VersionApi(BePersistencyInjector.injectVersionDao())
    }
}